lib/rules/selector-not-notation/__tests__/index.mjs

```diff
@@ -140,6 +152,40 @@ testRule({
 			endLine: 1,
 			endColumn: 12,
 		},
+		{
+			code: stripIndent`
+				/* a comment */
+				:not(::before), /* a comment */
+				:not(::before),
+				/* a comment */
+				:not(::before)
+				{}
+			`,
+			unfixable: true,
```

Comment:
Maybe another test for a case that is fixable?
Good to verify that we don't erase or mangle the comments.

Reply:
That would be covered by #7908.
see https://github.com/stylelint/stylelint/issues/7908#issuecomment-2287136983

Reply:
As @Mouvedia mentioned, this issue involves a different problem. 
Therefore, we can't add the test at this time.

See: [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcID0AqABAQzZAtjmAdgC5opIA6B8BERAFBgDQCUjaqm2EehJZl1WgxaNKHLLnzFSFKjXpNmaSsAC+lEIxAAzAJYAbGADkM+BCBgAPUwAdDAOjABnJ5vAQCegObnglNMogAE4AroZO5AhofgQBAZFOMIZgRBBBALTymbQYRLoekfCBTro4djCR-mjqBKpukJ66XgBiaTi55gBWTh5usDauiDHxIE5EAJ6G+rrEhYH6uTBjkaKxxRNTM0TpDd7pYxgEUBhBUHORC0RLRJW1IKpAA)

Reply:
No, this is directly related to this issue.

On line `78` you need to change:

```diff
- const selector = ruleNode.selector;
+ const selector = getRuleSelector(ruleNode);
```

Some examples of issues that currently exist:
- https://stylelint.io/demo/#N4Igxg9gJgpiBcID0AqABAZwC4E8A2MeAlgHZYC0URGAhgEYHkkwAeFxzmhMYWEATkwgUSwmliIQSaFEgA6JeKKwAKGgBoAlOrSo0NNJAC2RmGRnzFytVvUK9B46fOyFS4Tc1oFwAL4KQdRAAMyICADkaUwQQViiABwIAOjAMDEDwKVCAcxjgBTRvEH4AVwIMOQQ0fOlCooxuXgEhETEJKUr4eqIjRJhKgrR-El8MyBIcgDEBI3EYgCsMKQzYePTEGsLK7HxCUixOorxxGGxKu1rt3AIOCnGc8mwaEigafihDyuOsU4OQBVGviAA
- https://stylelint.io/demo/#N4Igxg9gJgpiBcCB2EAuAKAhgGgJTYHoAqAAgGdUBPAGxmoEslUBaKeszAI1uaRgA8WDPuTowwqCACdeaWakyp6EJCSIEAOkngoMOfFuIlMJSAFszMJms3bdWPCS3AAvlpDYQAM3q0AcpiWCCACgQAOtAB0YGRkHuAqPgDmwcBaJE4gUgCutGQaCCRpqhmZZGIS0vLyispIBfBl9GYRMAXpJG5ILvGQSMkAYtJmisEAVmQq8bBhcYjFGQUUNHSMqA2Z1IowFAXYHUtUtMIsfcnMFJhIUJhSUBsFW6g76yBaPS5AA

Reply:
Ah, I see. Sharp insight! I've applied the fix.

[Fix selector-not-notation autofix](https://github.com/stylelint/stylelint/pull/8074/commits/79ab762cde48f867928ac2cb9d4a2087f8d58746)

Reply:
Fixed. I've also left one more comment regarding a concern.

[Fix to ensure stylelint-disable-next-line works correctly](https://github.com/stylelint/stylelint/pull/8074/commits/fa9e135936ad886f41fd6dd58c50ddc42d7bb35f)